text/0000-documentation-related-metadata.md

```diff
@@ -0,0 +1,156 @@
+
+In addition, for existing Salesforce-authored modules there's a need to put component authors in control of this metadata. Currently a hardcoded list is maintained by a single team which is a roadblock for other teams creating new components or wishing to update the values.
+
+## Detailed design
+
+Key-value pairs will be placed as [frontmatter](https://github.com/jonschlinkert/gray-matter#what-does-this-do) (in YAML format) in the module markdown file that lives under `__docs__` in the module bundle. For example, this could be the content for `input/__docs__/input.md`:
+
+```markdown
+---
+category: Data Entry
+experience: All
+---
+
+descriptive documentation...
+```
+
+### Recognized Fields
+
+LWC itself will not restrict the set of fields a component author could add to the frontmatter, however only specific fields will be documented as recognized by the Salesforce platform. Unrecognized fields may be present but they will be ignored.
+
+The following keys will be recognized in the frontmatter:
+
+* __category__  `string` \
+    The main category for the module. Only one may be specified. Salesforce will maintain a list of recognized categories in public documentation and the VS Code extension for code completion/linting. For applications like the Component Library, values that match the whitelist will be displayed on the component’s reference page. Also see the filters on the homepage of the [Component Library](https://developer.salesforce.com/docs/component-library/overview/components) for other ways this information can be used.
+* __experience__  `string[]` \
+    One or more Salesforce experiences that the module works in. Salesforce will maintain a list of recognized experiences in public documentation and the VS Code extension for code completion/linting. For applications like the Component Library, values that match the whitelist will be displayed on the component’s reference page. Also see the filters on the homepage of the [Component Library](https://developer.salesforce.com/docs/component-library/overview/components) for other ways this information can be used.
+* __isSubComponent__  `boolean` \
+    This denotes that the component is only intended for use inside of another component, for example `lightning-breadcrumb` is only used inside of `lightning-breadcrumbs`. The Component Library may use this information for filtering options, to automatically cross-link pages, or to provide [component grouping](https://gus.lightning.force.com/lightning/r/a07B0000006HrUNIA0/view). Also this information could be used for linting.
+* __deprecated__  `boolean` \

```

Comment:
> Is there an existing way deprecation is already specified? 

I am not aware of a standard way to mark a component as deprecated.

> Does it make more sense to use the `@deprecated` tag from JSDoc? 

It makes sense to keep the `@deprecated` notation in JSDoc. Even if a module/component is not deprecated, some of their methods and properties might be.

> Also, should we have another field for the replacement component?

It is not uncommon that a deprecated module doesn't have a 1-1 replacement. Deprecated requires most of the time some free form text to indicate how to migrate, we can ask for deprecated modules to add a banner in the documentation explaining the migration strategy. We can also add another property called `deprecationMessage`  to have a standard way to surface in the component library and the different tools.

Reply:
I like the idea of `deprecationMessage`, as that can be used easier in tools like the editor LSPs. For more complicated scenarios I agree this could be supplemented with more information in the documentation itself.

When I talked with Alex and Ravi it came up that a deprecation flag might be added to the `.js-meta.xml` file. I'm not sure if it belongs there, but I agree the `@deprecated` tag works better than in the doc metadata should that can be a separate discussion.